src/sparsezoo/v2/helpers.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import List, Optional, Union
+
+from sparsezoo.v2 import Directory, File, NumpyDirectory, SampleOriginals
+
+
+def setup_model_directory(
+    output_dir: str,
+    training: Union[str, Directory],
+    deployment: Union[str, Directory],
+    onnx_model: Union[File, str],
+    sample_inputs: Union[str, NumpyDirectory],
+    sample_outputs: Union[
+        List[Union[str, NumpyDirectory]], str, NumpyDirectory, None
+    ] = None,
+    sample_labels: Union[Directory, str, None] = None,
+    sample_originals: Union[SampleOriginals, str, None] = None,
+    logs: Union[Directory, str, None] = None,
+    analysis: Union[File, str, None] = None,
+    benchmarks: Union[File, str, None] = None,
+    eval_results: Union[File, str, None] = None,
+    model_card: Union[File, str, None] = None,
+    recipes: Union[List[Union[str, File]], str, File, None] = None,
+) -> None:
+    """
+
+    The function takes Files and Directories that are expected by the
+    ModelDirectory (some Files/Directories are mandatory, some are optional),
+    and then creates a new directory where the files are being copied to.
+    The format of the new directory adheres to the structure expected by the
+    `ModelDirectory` class factory methods.
+
+
+    :params output_dir: path to the target directory
+    :params training: pointer (path or File) to the training directory
+    :params deployment: pointer (path or File) to the deployment directory
+    :params onnx_model: pointer (path or File) to the model.onnx file
+    :params sample_inputs: pointer (path or File) to the sample_inputs directory
+    :params sample_outputs: pointer (path or File) to the sample_outputs directory
+            (also supports list of paths or Files)
+    :params sample_labels: pointer (path or File) to the sample_labels directory
+    :params sample_originals: pointer (path or File) to the sample_originals directory
+    :params logs: pointer (path or File) to the logs directory
+    :params analysis: pointer (path or File) to the analysis.yaml file
+    :params benchmarks: pointer (path or File) to the benchmarks.yaml file
+    :params eval_results: pointer (path or File) to the eval.yaml file
+    :params model_card: pointer (path or File) to the model.md file
+    :params recipes: pointer (path or File) to the recipe.yaml file
+            (also supports list of paths or Files)
+    """
+    # create new directory
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+
+    # fetch the function arguments as a dictionary
+    files_dict = copy.deepcopy(locals())
+    del files_dict["output_dir"]
+
+    for name, file in files_dict.items():
+        if file is None:
+            logging.debug(f"File {name} not provided, skipping...")
+        else:
+            if isinstance(file, str):
+                file = _create_file_from_path(file)
+            elif isinstance(file, list) and isinstance(file[0], str):
```

Comment:
What is `isinstance(file[0], str)` checking for?

This feels like an instance where an else catch would be helpful